torch/utils/cpp_extension.py

```diff
@@ -85,8 +85,9 @@ def _find_cuda_home():
 
                               !! WARNING !!
 '''
-CUDA_HOME = _find_cuda_home()
-CUDNN_HOME = os.environ.get('CUDNN_HOME') or os.environ.get('CUDNN_PATH')
+HIP_COMP = os.path.exists('/opt/rocm')
```

Comment:
This feels wrong. The presence of `/opt/rocm` folder doesn't mean that I didn't want CUDA. You should be querying whether or not the PyTorch build was a CUDA or ROCM build and act accordingly (or, in a pinch, toggle this by environment variable.)

Reply:
I see your point, I'd love for a reliable way to detect HIP compilation, I am not sure there is one currently. So maybe environment variable...

TBF, the build scripts rely on detecting `/opt/rocm` to switch the HIP build on currently.

Reply:
So this should be easy by testing `version.hip is not None` now...